src/trace/trace.c

```diff
@@ -220,3 +221,37 @@ void trace_init(struct sof *sof)
 	dcache_writeback_invalidate_region((void *)MAILBOX_TRACE_BASE,
 					   MAILBOX_TRACE_SIZE);
 }
+
+void trace_ratelimit_init(struct sof *sof)
+{
+	sof->trace->ratelimit_timeout = clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK,
+						1) / 1000 * TRACE_RATELIMIT_INTERVAL;
+}
+
+bool trace_ratelimit_time(struct trace_ratelimit *trtl)
+{
+	uint64_t now = platform_timer_get(timer_get());
+	struct trace *trace = trace_get();
+	bool ret = true;
+
+	if (now - trtl->last < trace->ratelimit_timeout) {
+		if (trtl->count++ > TRACE_RATELIMIT_BURST)
+			ret = false;
+	} else {
+		trtl->missed = trtl->count;
+		trtl->count = 0;
+	}
+
+	if (ret)
+		trtl->last = now;
+
+	return ret;
+}
+
+bool trace_ratelimit_freq(struct trace_ratelimit *trtl)
+{
+	if (trtl->count && !(TRACE_RATELIMIT_FREQ & trtl->count))
```

Comment:
ditto here.

Reply:
@lgirdwood done. Interesting, CI isn't restarting automatically. I don't expect it to be smart enough to notice, that only comments changed from the previous version... We can kick it manually or rely on its apparent intelligence.